catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/BqsConverter.java

```diff
@@ -186,11 +191,13 @@ public void exitQuery(BqsParser.QueryContext ctx) {
                 String operHash = nestedOperatorStack.pop();
                 List<Filter> filters = filterBy.get(operHash);
 
-                if (currFilter == null) {
-                    currFilter = filterBuilder.anyOf(filters);
-                } else {
-                    filters.add(currFilter);
-                    currFilter = filterBuilder.anyOf(filters);
+                if (filters != null && !filters.isEmpty()) {
+                    if (currFilter == null) {
+                        currFilter = filterBuilder.anyOf(filters);
+                    } else {
+                        filters.add(currFilter);
+                        currFilter = filterBuilder.anyOf(filters);
```

Comment:
nitpick - only do the filterBuilder.anyOf(filters) once:
if (currentFilter !- null) {
  filters.add(currFilter)
}
currFilter = filterBuilder.anyOf(filters);